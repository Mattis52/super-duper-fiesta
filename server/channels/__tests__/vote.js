jest.mock('../../models/user');
jest.mock('../../models/vote', () => ({
  haveIVoted: async () => false,
  createVote: () => ({
    save: async () => jest.fn(),
  }),
}));
jest.mock('../../models/issue', () => ({
  getIssueById: async issueId => ({
    _id: issueId,
    active: true,
    genfors: '1',
  }),
}));
jest.mock('../../models/meeting', () => ({
  getActiveGenfors: async () => ({
    id: '1',
  }),
  getGenfors: async id => ({
    id,
  }),
}));
jest.mock('../../utils');
const { submitRegularVote } = require('../vote');
const { broadcast, emit } = require('../../utils');

const generateSocket = user => ({
  request: {
    user: {
      _id: '123',
      onlinewebId: '123',
      name: 'Namy',
      completedRegistration: false,
      permissions: 5,
      genfors: '1',
      ...user,
    },
    headers: {
      cookie: {
        passwordHash: 'hashy',
      },
    },
  },
});

const generateData = () => ({
  issue: 1,
  alternative: 1,
});

describe('submitRegularVote', () => {
  it('returns error when not registered', async () => {
    await submitRegularVote(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
  });
  it('does something', async () => {
    await submitRegularVote(generateSocket({ completedRegistration: true }), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
  });
});
