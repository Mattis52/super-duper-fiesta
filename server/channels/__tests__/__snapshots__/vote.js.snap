// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitRegularVote does something 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": undefined,
      "option": undefined,
      "question": undefined,
      "user": undefined,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "VOTED_STATE",
    Object {
      "voted": true,
    },
  ],
]
`;

exports[`submitRegularVote returns error when not registered 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": false,
          "genfors": "1",
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {},
    Object {
      "error": "Du er ikke registert",
    },
  ],
]
`;
